### PR TITLE
[#115931361] Add haproxy with option of adding raw config to frontend

### PR DESCRIPTION
**Pull request by keymon**
## What

We want to serve [HSTS
headers](https://en.wikipedia.org/wiki/HTTP_Strict_Transport_Security) for all
HTTPS requests to the apps domains, but it will safeguard existing users from
being MITMed over insecure connections and it will improve the user experience
when they click on a hostname that doesn't have a protocol.

We want to leave open the option of able to override these headers from
the tenant application if they wish.

We decided to use haproxy in front of the gorouter to do the job for the time
being, but the [current haproxy job from `cf-release`](https://github.com/cloudfoundry/cf-release/tree/master/jobs/haproxy/)
does not support adding custom config.
## Context

We decided:
- create a custom release copying the haproxy job
- allow define raw HAProxy config in the manifest.
## How to test?

This reelase can be tested as part of the PR in paas-cf https://github.com/alphagov/paas-cf/pull/221,
which currently pulls and uses this branch.

You can review both PR at the same time.
## Additional notes

After merging this PR, you should:
- tag the master commit with `0.0.1`
- change the paas-cf pipeline `create-bosh-cloudfoundry` to filter this
  git repository based on the tag with `tag_filter`
## Who?

Anyone but @keymon or @HenryTK

### CODE REVIEW

File: src/haproxy/services/utils.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
+source /var/vcap/packages/haproxy/utils.sh
```
Comment (mtekel) [inline src/haproxy/services/utils.sh]: this script seems to do things with Warden. How is this related to haproxy? Is this the script that was added from common package? Is the only functionality it provides sourcing real haproxy script?

File: config/final.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (mtekel) [inline config/final.yml:1]: Looks like this is taken from cf-release. I think we should remove this in subsequent commits (if we don't intend to make final release of this job or adjust to point to our bucket), because now it points to the same bucket as CF-release.

**Comment by mtekel** [comment]
Does this release have any external dependencies? E.g. pcre and haproxy packages? What happens if those are changed? Does it make sense to turn this into standalone release and make final release?

File: config/final.yml (hunk 2 above)
Comment (keymon) [inline config/final.yml:1]: Yes, it does, it was on purpose because we want to use the same package than cf-release.

**Comment by keymon** [comment]
It depends on cf-release packages.

Do a final release was initially out of scope. In case of doubt, we can have a tech-talk about the scope of this story.

File: src/haproxy/services/utils.sh (hunk 1 above)
Comment (keymon) [inline src/haproxy/services/utils.sh]: I will remove this script.

**Comment by mtekel** [comment]
works fine.
